src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -103,10 +103,12 @@ protected override void CheckBase(BindingDiagnosticBag diagnostics)
             }
 
             // you need to know all bases before you can ask this question... (asking this causes a cycle)
-            if (this.IsGenericType && !baseContainsErrorTypes && this.DeclaringCompilation.IsAttributeType(localBase))
+            if (this.IsGenericType
+                && !baseContainsErrorTypes
+                && this.DeclaringCompilation.IsAttributeType(localBase)
+                && MessageID.IDS_FeatureGenericAttributes.GetFeatureAvailabilityDiagnosticInfo(this.DeclaringCompilation) is { } diagnosticInfo)
```

Comment:
>&& MessageID.IDS_FeatureGenericAttributes.GetFeatureAvailabilityDiagnosticInfo(this.DeclaringCompilation) is { } diagnosticInfo) [](http://example.com/codeflow?start=16&length=128)

Perhaps calling `MessageID.IDS_FeatureGenericAttributes.CheckFeatureAvailability(diagnostics, DeclaringCompilation, baseLocation);` instead since it's slightly shorter.

Reply:
Sounds good.